homeassistant/components/mailbox/__init__.py

```diff
@@ -0,0 +1,256 @@
+
+        hass.bus.async_listen(EVENT, _mailbox_updated)
+
+    @property
+    def state(self):
+        """Return the state of the binary sensor."""
+        return str(len(self.mailbox.get_messages()))
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self.mailbox.name
+
+
+class Mailbox(object):
+    """Represent an mailbox device."""
+
+    def __init__(self, hass, name):
+        """Initialize mailbox object."""
+        self.hass = hass
+        self.name = name
+
+    def update(self):
+        """Send event notification of updated mailbox."""
+        self.hass.bus.async_fire(EVENT)
+
+    def get_media_type(self):
+        """Return the supported media type."""
+        raise NotImplementedError()
+
+    def get_media(self, msgid):
+        """Return the media blob for the msgid."""
+        raise NotImplementedError()
+
+    def get_messages(self):
+        """Return a list of the current messages."""
+        raise NotImplementedError()
+
+    def delete(self, msgids):
+        """Delete the specified messages."""
+        raise NotImplementedError()
+
+
+class StreamError(Exception):
+    """Media streaming exception."""
+
+    pass
+
+
+class MailboxView(HomeAssistantView):
+    """Base mailbox view."""
+
+    def __init__(self, mailboxes):
+        """Initialize a basic mailbox view."""
+        self.mailboxes = mailboxes
+
+    def get_mailbox(self, platform):
+        """Retrieve the specified mailbox."""
+        for mailbox in self.mailboxes:
+            if mailbox.name == platform:
+                return mailbox
+        return None
+
+
+class MailboxPlatformsView(MailboxView):
+    """View to return the list of mailbox platforms."""
+
+    url = "/api/mailbox/platforms"
+    name = "api:mailbox:platforms"
+
+    @asyncio.coroutine
+    def get(self, request):
+        """Retrieve list of platforms."""
+        platforms = []
+        for mailbox in self.mailboxes:
+            platforms.append(mailbox.name)
+        return self.json(platforms)
+
+
+class MailboxMessageView(MailboxView):
+    """View to return the list of messages."""
+
+    url = "/api/mailbox/messages/{platform}"
+    name = "api:mailbox:messages"
+
+    @asyncio.coroutine
+    def get(self, request, platform):
+        """Retrieve messages."""
+        mailbox = self.get_mailbox(platform)
+        if mailbox is None:
+            return web.Response(status=401)
+        return self.json(mailbox.get_messages())
+
+
+class MailboxDeleteView(MailboxView):
+    """View to delete selected messages."""
+
+    url = "/api/mailbox/delete/{platform}"
+    name = "api:mailbox:delete"
+
+    @asyncio.coroutine
+    def post(self, request, platform):
+        """Delete items."""
+        mailbox = self.get_mailbox(platform)
+        if mailbox is None:
+            return web.Response(status=401)
+        try:
+            data = yield from request.json()
```

Comment:
Any JSON received from the client should be validated before being passed on. But if you go restful (as I commented), you won't need this.